domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -430,15 +431,15 @@ public void closeClient(Handler<AsyncResult<Void>> whenDone) {
    */
   public static void closeAllClients() {
     @SuppressWarnings("rawtypes")
-    List<Future> list = new ArrayList<>(connectionPool.size());
+    List<Promise<Object>> list = new ArrayList<>(connectionPool.size());
     // copy of values() because closeClient will delete them from connectionPool
     for (PostgresClient client : connectionPool.values().toArray(new PostgresClient [0])) {
-      Future<Object> future = Future.future();
-      list.add(future);
-      client.closeClient(f -> future.complete());
+      Promise<Object> promise = Promise.promise();
+      list.add(promise);
+      client.closeClient(f -> promise.complete());
     }
 
-    CompositeFuture.join(list);
+    CompositeFuture.join(list.stream().map(Promise::future).collect(Collectors.toList()));
   }
```

Comment:
```
    List<Future> list = new ArrayList<>(connectionPool.size());
    // copy of values() because closeClient will delete them from connectionPool
    for (PostgresClient client : connectionPool.values().toArray(new PostgresClient [0])) {
      Promise<Object> promise = Promise.promise();
      list.add(promise.future());
      client.closeClient(f -> promise.complete());
    }

    CompositeFuture.join(list);
```